docs/40-troubleshooting.md

```diff
@@ -1,4 +1,47 @@
-# Getting logs without Azure Portal
+## Debugging and Troubleshooting
+
+### After creating my cluster, my nodes go to a 'startTaskFailed' state. Why?
+The most common case for this is that there was an issue with package installation or the custom script failed to run. To troubleshoot this you can simply download the output logs from the node.
+
+```r
+cluster <- makeCluster('myConfig.json')
+...
+# Get standard error file
+getClusterFile(cluster, "tvm-1170471534_2-20170829t072146z", "stderr.txt", downloadPath = "pool-errors.txt")
+
+# Get standard log file
+getClusterFile(cluster, "tvm-1170471534_2-20170829t072146z", "stderr.txt", downloadPath = "pool-errors.txt")
+```
+
+### My job never starts running. How can I troubleshoot this issue?
+This is often caused by the node not being in a good state. Take a look at the state of the nodes in the cluster to see if any of there are and nodes in an error or failed state. If not node is in a startTaskFailed state follow the instructions above. If the node is in an 'unknown' or 'unusable' state you may need to manually reboot the node.
+
+```r
+# reboot a node
+rAzureBatch::rebootNode('<my_cluster_id>', '<my_node_id>')
+```
+
+### Why do some packages fail with the following error?
+```sh
+ERROR: compilation failed for package '__PACKAGE__'
+* removing '/usr/lib64/microsoft-r/3.3/lib64/R/library/__PACKAGE__'
+```
+
+This issue is due to certain compiler flags not available in the default version of R used by doAzureParallel. In order to get around this issue you can add the following commands to the command line in the cluster configuration to make sure R has the right compiler flags set.
```

Comment:
i think we should add a link to the term "command line" to the command line docs https://github.com/Azure/doAzureParallel/blob/master/docs/30-customize-cluster.md#running-commands-when-the-cluster-starts